otherlibs/unix/unixsupport.h

```diff
@@ -66,14 +66,24 @@ struct filedescr {
 
 extern value caml_win32_alloc_handle(HANDLE);
 extern value caml_win32_alloc_socket(SOCKET);
+
+#define NO_CRT_FD (-1)
+#define GETTING_CRT_FD (-2)
```

Comment:
Swap these two around, so `atomic_load(&CRT_fd_val(handle)) > NO_CRT_FD` means that `caml_win32_CRT_fd_of_filedescr` has already been used or is in the process of allocating a fd.

```suggestion
#define NO_CRT_FD (-2)
#define GETTING_CRT_FD (-1)
```